nostr-client-plus/src/bin/aggregator.rs

```diff
@@ -306,28 +306,40 @@ async fn run() -> Result<()> {
             let raw_data_id = hex::encode(new_payload.header.raw_data_id.hash());
 
             // First fetch all existing tasks
-            let saved_tasks: Result<String, RedisError> =
-                (&mut redis_con_task).get(raw_data_id.clone());
-            let mut tasks: Vec<AssignerTask> = match saved_tasks {
-                Ok(val) => {
-                    match from_str(val.as_str()) {
+            let mut tasks: Vec<AssignerTask> = Vec::new();
+            // Only try to fetch redis record if the key exists
+            match (&mut redis_con_task).exists(raw_data_id.clone()) {
+                Ok(true) => {
+                    let saved_tasks: Result<String, RedisError> =
+                        (&mut redis_con_task).get(raw_data_id.clone());
+                    tasks = match saved_tasks {
                         Ok(val) => {
-                            val
-                        }
+                            match from_str(val.as_str()) {
+                                Ok(val) => {
+                                    val
+                                }
 
+                                Err(err) => {
+                                    tracing::error!("Failed parse tasks, error: {}", err);
+                                    // Skip this task since the stored tasks are corrupted
+                                    continue;
+                                }
+                            }
+                        }
                         Err(err) => {
-                            tracing::error!("Failed parse tasks, error: {}", err);
-                            // Skip this task since the stored tasks are corrupted
+                            tracing::error!("Failed to fetch tasks from redis for data: {}, error: {}", raw_data_id.clone(), err);
+                            // Skip this task since something wrong with redis
                             continue;
                         }
-                    }
+                    };
                 }
                 Err(err) => {
-                    tracing::error!("Failed to fetch tasks from redis for data: {}, error: {}", raw_data_id.clone(), err);
+                    tracing::error!("Failed to check if tasks exits from redis for data: {}, error: {}", raw_data_id.clone(), err);
                     // Skip this task since something wrong with redis
                     continue;
                 }
-            };
+                _ => {}
```

Comment:
No need to use the "catch-all" branch, there is only another case you did not cover, and it's `Ok(false)` so, please, be explicit.

That said, I do not understand why we need to `match` twice: if we just need to check for existence, then just the first match, but if you need to retrieve the key, then just call `get` and then handle the result.
If there is already a vector, good. If there is nothing, then create an empty vec and assign to `tasks`. If there is an error, do what you already did.

So, bottom line is, do not check for existence and then retrieve, just retrieve.

I could not test the `get` method because I have no redis, but because it's just a wrapper for Redis' `GET` and `MGET`, I would expect the same behaviour.

Reply:
In case the key not exists, it returns an error, so just want to be explicit since distinguishing between errors can be annoying.